release-process.md

```diff
@@ -1,54 +1,131 @@
+- Does production have required configuration in place?
+- In the timing of this release appropriate?
+  - Typically we avoid end-of-day or end-of-week releases because a bug could cause unnecessary off-hours intervention.
+  - If a change is urgent (e.g. website is down), we will sometimes decide to move ahead with this even if the timing is not a typically ideal one.
+
+## Overview
 
 In addition to creating a code snapshot for production, the release process
 triggers code deployments.  There is already tooling that deploys commits to
-``master/main`` to the ``<projectname>-ci`` server. The release process triggers
+``master/main`` to the ``<projectname>-ci`` server (this may not exist in some projects). The release process triggers
 deployments to ``<projectname>-rc`` for acceptance testing.  When
 ``release-candidate`` is merged to ``release``, the application is deployed
 to the production server.
+## Release Lifecycle
+
+### Development
+
+The development process is the same regardless of the type of project:
 
-## Steps
-### 1. Check-out a current ``master/main`` branch
-Verify that ``master/main`` contains the commits intended for the release.  If
-there isn't already an issue in GitHub for the release, create one.
-### 2. Create a ``release-candidate`` branch and hard reset it to ``master/main``
-The ``release-candidate`` branch is hard reset to ``master/main`` since for the
-purposes of the release process, its state on the remote is irrelevant.
-### 4. Generate release notes
-All of our projects contain a file for release notes in the project
-root directory.  This file is named ``RELEASE.rst``.  For each release it is
-updated with a section titled "Release <version-number>" containing a list
-of the commits in the release.  
-### 5. Update version numbers
-Project version numbers are typically stored in ``settings.py`` and/or
-``setup.py`` depending on the application's architecture and purpose.
-### 6. Commit updates and push ``release-candidate`` branch
-### 7. Generate release notes with checkboxes
-### 8. Open PR to merge ``release-candidate`` branch into ``release`` branch
-### 9. Merge PR once developers verify their commits (manual step)
-### 10. Tag the release
-### 11. Merge the ``release`` branch into the ``master/main`` branch and push
-### 12. Send email notifications (manual step)
-
-
-## Release Version Numbers
-Our releases are identified by a version number that follows the
+  - Code starts out in a pull request which is reviewed by other team members (see [Code Review](code-review.md)).
+  - Then the pull request is merged to the `main` branch (or `master` for legacy projects).
+
+### Process
+
+```mermaid
+sequenceDiagram
+  actor eng as Software Engineers
+  participant branchmain as main/master
+
+  participant branchrc as release-candidate
+  participant envrc as Release Candidate
+
+  participant branchrelease as release
+  participant envprod as Production
```

Comment:
Minor suggestion... Casing and context made it pretty clear, but still took me a second:
```suggestion
  participant branchrc as "release-candidate <br> branch"
  participant envrc as "Release Candidate <br> Environment"

  participant branchrelease as "release <br> branch"
  participant envprod as "Production <br> Environment"
```

(*I'm reasonably certain the syntax above will work, but I have not tried it with this particular diagram..*)